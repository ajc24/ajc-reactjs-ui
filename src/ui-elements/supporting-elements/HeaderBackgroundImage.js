import React from 'react';
import PropTypes from 'prop-types';
import '../../styling/ui-styles.css';

class HeaderBackgroundImage extends React.Component {
  constructor(props) {
    super(props);
  }

  render() {
    const imageStyling = 'ajc-header-image';
    return (
      <React.Fragment>
        <img id={this.props.id} className={imageStyling} src={this.props.imageSrc} role="presentation" />
      </React.Fragment>
    );
  }
}
HeaderBackgroundImage.propTypes = {
  /** The unique identifier for the component. */
  id: PropTypes.string.isRequired,
  /** The image to be displayed within the component. */
  imageSrc: PropTypes.oneOfType([ PropTypes.string, PropTypes.object ]).isRequired,
};
export default HeaderBackgroundImage;
