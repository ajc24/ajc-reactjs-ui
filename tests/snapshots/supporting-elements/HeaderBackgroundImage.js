import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import HeaderBackgroundImage from '../../../src/ui-elements/supporting-elements/HeaderBackgroundImage';
import testImage from '../../../stories/images/storybook-header-bg.png';

describe('HeaderBackgroundImage', () => {
  describe('Default props and rendering - Image src as an import of image data', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <div role="banner">
          <HeaderBackgroundImage id="test-image-id" imageSrc={testImage} />
        </div>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Default props and rendering - Image src as a string', () => {
    let jsonSnapshot;

    beforeAll(() => {
      const testImagePath = '../../../stories/images/storybook-header-bg.png';
      jsonSnapshot = TestDev.createSnapshot(
        <div role="banner">
          <HeaderBackgroundImage id="test-image-id" imageSrc={testImagePath} />
        </div>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
