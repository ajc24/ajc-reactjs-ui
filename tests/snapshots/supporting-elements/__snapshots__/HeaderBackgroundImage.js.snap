// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderBackgroundImage Default props and rendering - Image src as a string verifies the snapshot for the component 1`] = `
<div
  role="banner"
>
  <img
    className="ajc-header-image"
    id="test-image-id"
    onLoad={[Function]}
    role="presentation"
    src="../../../stories/images/storybook-header-bg.png"
  />
</div>
`;

exports[`HeaderBackgroundImage Default props and rendering - Image src as an import of image data verifies the snapshot for the component 1`] = `
<div
  role="banner"
>
  <img
    className="ajc-header-image"
    id="test-image-id"
    onLoad={[Function]}
    role="presentation"
    src={Object {}}
  />
</div>
`;
